src/instructlab/rag/rag_configuration.py

```diff
@@ -0,0 +1,141 @@
+# Standard
+from typing import Any, Optional
+import logging
+import os
+
+# Third Party
+from pydantic import BaseModel, ConfigDict
+import click
+
+# First Party
+from instructlab.configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+
+def rag_options(command):
+    """Wrapper to apply common options."""
+    command = click.option(
+        "--rag",
+        "is_rag",
+        is_flag=True,
+        envvar="ILAB_RAG",
+        help="To enable Retrieval-Augmented Generation",
+    )(command)
+    command = click.option(
+        "--document-store-type",
+        default="milvuslite",
+        envvar="ILAB_DOCUMENT_STORE_TYPE",
+        type=click.STRING,
+        help="The document store type, one of: `milvuslite`, `milvus`.",
+    )(command)
+    command = click.option(
+        "--document-store-uri",
+        default="embeddings.db",
+        envvar="ILAB_DOCUMENT_STORE_URI",
+        type=click.STRING,
+        help="The document store URI",
+    )(command)
+    command = click.option(
+        "--document-store-collection-name",
+        default="IlabEmbeddings",
+        envvar="ILAB_DOCUMENT_STORE_COLLECTION_NAME",
+        type=click.STRING,
+        help="The document store collection name",
+    )(command)
+    command = click.option(
+        "--retriever-top-k",
+        default=10,
+        envvar="ILAB_RETRIEVER_TOP_K",
+        type=click.INT,
+    )(command)
+    command = click.option(
+        "--retriever-embedder-model-dir",
+        default=lambda: DEFAULTS.MODELS_DIR,
```

Comment:
things like this for example should be reading from the config, we try not to specify defaults at this level